compiler/rustc_typeck/src/check/check.rs

```diff
@@ -87,8 +87,26 @@ pub(super) fn check_fn<'a, 'tcx>(
 
     let declared_ret_ty = fn_sig.output();
 
-    let revealed_ret_ty =
-        fcx.instantiate_opaque_types_from_value(fn_id, declared_ret_ty, decl.output.span());
+    let feature = match tcx.hir().get(fn_id) {
+        Node::Item(hir::Item { kind: ItemKind::Fn(..), .. }) |
```

Comment:
I'd like to see comments on each arm with an example of the corresponding rust code